source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md

```diff
@@ -0,0 +1,114 @@
+---
+title: How to build new OpenStack packages
+author: chandankumar
+tags: dlrn,rdo,openstack,packaging,centos
+date: 2016-07-15 17:37:12 UTC
+---
+
+Building new OpenStack packages for RDO is always tough. Let's use DLRN to make our life simpler.
+
+DLRN is the RDO Continuous Delivery platform that pulls upstream git, rebuild them as RPM using template spec files, and ships them in
+repositories consumable by CI (e.g upstream puppet/Tripleo/packstack CI).
+
+We can use DLRN to build a new RDO python package before sending them for package review.
+
+## Install DLRN
+
+[1.] Install required dependencies for DLRN on Fedora/CentOS system:
+
+    $ sudo yum install git createrepo python-virtualenv mock gcc \
+                  redhat-rpm-config rpmdevtools libffi-devel \
+                  openssl-devel
+
+
+[2.] Create a virtualenv and activate it
+
+    $ virtualenv dlrn-venv
+    $ source dlrn-venv/bin/activate
+
+[3.] Clone DLRN git respository from github
+
+    $ git clone https://github.com/openstack-packages/DLRN.git
+
+[4.] Install the required python dependencies for DLRN
+
+    $ cd DLRN
+    $ pip install -r requirements.txt
+
+[5.] Install DLRN
+
+    $ python setup.py develop
+
+[6.] Add the user you intend to run as to the mock group:
+
+    $ sudo usermod -a -G mock $USER
+    $ newgrp mock
+    $ newgrp $USER
```

Comment:
unneeded, mock package does it for you.
http://pkgs.fedoraproject.org/cgit/rpms/mock.git/tree/mock.spec#n153